unit_testing/unit_test.py

```diff
@@ -662,3 +665,56 @@
     print(str(sec) + chr(subsec) + " X - Altimetry quick plot not saved")
 
 plt.close('all')
+
+#################################################
+## ( 8 ) TIDEGAUGE Methods                     ##
+#################################################
+sec = sec+1
+subsec = 96
+
+#-----------------------------------------------------------------------------#
+# ( 8a ) Load in GESLA tide gauge files from directory                        #
+#                                                                             #
+subsec = subsec+1
+
+try:
+    date0 = datetime.datetime(2010,1,1)
+    date1 = datetime.datetime(2010,12,1)
+    tg = coast.TIDEGAUGE(dn_tidegauge, date0, date1)
+    
+    # Check length of dataset_list is correct and that
+    if len(tg.dataset_list) == 9:
+        print(str(sec) + chr(subsec) + " OK - Tide gauges loaded")
+except:
+    print(str(sec) + chr(subsec) +' FAILED.')
+
+#-----------------------------------------------------------------------------#
+# ( 8b ) TIDEGAUGE map plot                                                   #
+#                                                                             #
+subsec = subsec+1
+
+try:
+    f,a = tg.plot_map()
+    f.savefig(dn_fig + 'tidegauge_map.png')
+    print(str(sec) + chr(subsec) + " OK - Tide gauge map plot saved")
+except:
+    print(str(sec) + chr(subsec) +' FAILED.')
+    
+plt.close('all')
+
+#-----------------------------------------------------------------------------#
+# ( 8c ) TIDEGAUGE Time series plot                                           #
+#                                                                             #
+subsec = subsec+1
+
+
+try:
+    f,a = tg.plot_timeseries(0)
```

Comment:
The Fukaura example is OK but is there one with more than just "Correct" data? (It doesn't show off your legend well)

Reply:
This is easier said than done. The GESLA data files contain the option to have a few different QC flags, but they aren't all that common in the actual data..